egs/wsj/s5/steps/nnet3/train_discriminative.sh

```diff
@@ -307,24 +303,25 @@ while [ $x -lt $num_iters ]; do
       nnet3-am-copy --set-raw-nnet=- $dir/$x.mdl $dir/$[$x+1].mdl || exit 1;
 
     rm $nnets_list
+    [ ! -f $dir/$[$x+1].mdl ] && echo "$0: Did not create $dir/$[$x+1].mdl" && exit 1;
+    if [ -f $dir/$[$x-1].mdl ] && $cleanup && \
+       [ $[($x-1)%$keep_model_iters] -ne 0  ] && \
```

Comment:
I don't think this has very good defaults... $keep_model_iters defaults to 1 which means all models are kept initially.
And I'm concerned that the code is too complicated... users will assume that these keep_model_iters are kept permanently, not deleted at the end.
I think it would be better to delete the [x-5]'th model (so we have 5 models for debugging purposes), and have the cleanup stage at the end delete the last 5 models... and just hardcode this 5 in the code, I don't see that anyone would ever want to configure it.  Right now I think it's configurable to a confusing extent.